go/tools/builders/compilepkg.go

```diff
@@ -178,7 +178,12 @@ func compileArchive(
 	defer cleanup()
 
 	if len(srcs.goSrcs) == 0 {
-		emptyPath := filepath.Join(workDir, "_empty.go")
+		// We need to run the compiler to create a valid archive, even if there's
+		// nothing in it. GoPack will complain if we try to add assembly or cgo
+		// objects.
+		// _empty.go needs to be in a deterministic location (not tmpdir) in order
+		// to ensure deterministic output
+		emptyPath := filepath.Join(filepath.Dir(outPath), "_empty.go")
```

Comment:
This strategy is already being used here https://github.com/bazelbuild/rules_go/blob/master/go/tools/builders/compile.go#L98